pkg/function/delete_data_using_kopia_server.go

```diff
@@ -0,0 +1,215 @@
+		return nil, err
+	}
+	if err = Arg(args, kankopia.KopiaUserPassphraseArg, &userPassphrase); err != nil {
+		return nil, err
+	}
+
+	if err = Arg(args, kankopia.KopiaTLSCertSecretDataArg, &cert); err != nil {
+		return nil, err
+	}
+
+	fingerprint, err := kankopia.ExtractFingerprintFromCertificateJSON(cert)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to fetch Kopia API Server Certificate Secret Data from blueprint")
+	}
+
+	username := tp.RepositoryServer.Username
+	hostname, userAccessPassphrase, err := getHostNameAndUserPassPhraseFromRepoServer(userPassphrase)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get hostname/user passphrase from Options")
+	}
+
+	cli, err := kube.NewClient()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to create Kubernetes client")
+	}
+
+	repositoryServerService, err := cli.CoreV1().Services(tp.RepositoryServer.Namespace).Get(context.Background(), tp.RepositoryServer.ServerInfo.ServiceName, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.New("Unable to find Service Details for Repository Server")
+	}
+	repositoryServerServicePort := strconv.Itoa(int(repositoryServerService.Spec.Ports[0].Port))
+	serverAddress := "https://" + tp.RepositoryServer.ServerInfo.ServiceName + "." + tp.RepositoryServer.Namespace + ".svc.cluster.local:" + repositoryServerServicePort
+
+	return deleteDataFromServer(
+		ctx,
+		cli,
+		hostname,
+		image,
+		deleteDataJobPrefix,
+		namespace,
+		serverAddress,
+		fingerprint,
+		snapID,
+		username,
+		userAccessPassphrase,
+	)
+}
+
+func deleteDataFromServer(
+	ctx context.Context,
+	cli kubernetes.Interface,
+	hostname,
+	image,
+	jobPrefix,
+	namespace,
+	serverAddress,
+	fingerprint,
+	snapID,
+	username,
+	userPassphrase string,
+) (map[string]any, error) {
+
+	options := &kube.PodOptions{
+		Namespace:    namespace,
+		GenerateName: jobPrefix,
+		Image:        image,
+		Command:      []string{"bash", "-c", "tail -f /dev/null"},
+	}
+
+	kankopia.SetLabelsToPodOptionsIfRequired(options)
+	kankopia.SetAnnotationsToPodOptionsIfRequired(options)
+	kankopia.SetResourceRequirementsToPodOptionsIfRequired(options)
+
+	pr := kube.NewPodRunner(cli, options)
+	podFunc := deleteDataFromServerPodFunc(
+		cli,
+		hostname,
+		namespace,
+		serverAddress,
+		fingerprint,
+		snapID,
+		username,
+		userPassphrase,
+	)
+	return pr.Run(ctx, podFunc)
+}
+
+func deleteDataFromServerPodFunc(
+	cli kubernetes.Interface,
+	hostname,
+	namespace,
+	serverAddress,
+	fingerprint,
+	snapID,
+	username,
+	userPassphrase string,
+) func(ctx context.Context, pod *corev1.Pod) (map[string]any, error) {
+	return func(ctx context.Context, pod *corev1.Pod) (map[string]any, error) {
+		// Wait for pod to reach running state
+		if err := kube.WaitForPodReady(ctx, cli, pod.Namespace, pod.Name); err != nil {
+			return nil, errors.Wrap(err, "Failed while waiting for Pod: "+pod.Name+" to be ready")
+		}
```

Comment:
```suggestion
	return func(ctx context.Context, pod *corev1.Pod) (map[string]any, error) {
		if err := kube.WaitForPodReady(ctx, cli, pod.Namespace, pod.Name); err != nil {
			return nil, errors.Wrap(err, "Failed while waiting for Pod: "+pod.Name+" to be ready")
		}
```

I think function name `kube.WaitForPodReady` conveys the message clearly.

Reply:
Will remove it !